components/global/Nav/Nav.tsx

```diff
@@ -15,36 +17,89 @@ const isFirstBranchMatched = (pagePath: string, href: string) => {
+  const activeElement = React.useRef<Element>()
+  const { pathname } = useRouter()
+  const { id, expanded } = useNavigationState()
+  const [status, setStatus] = React.useState<NavStatus>('closed')
+
+  React.useEffect(() => {
+    if (expanded) {
+      activeElement.current = document.activeElement
+
+      if (navRef.current) {
+        navRef.current.focus()
+      }
+    } else if (activeElement.current) {
+      const active = activeElement.current
+
+      if (isFocusableElement(active)) {
+        active.focus()
+      }
+      activeElement.current = undefined
+    }
+  }, [expanded, activeElement])
+
+  React.useEffect(() => {
+    function keyPressHandler(evt: KeyboardEvent) {
+      if (evt.key === 'Escape') {
+        dispatch('close')
+        setStatus('closing')
+      }
+    }
+
+    if (expanded) {
+      document.addEventListener('keydown', keyPressHandler)
+    }
+
+    return function navKeyboardCleanup() {
+      document.removeEventListener('keydown', keyPressHandler)
+    }
+  }, [expanded, dispatch])
+
+  React.useEffect(() => {
+    const containerCurrent = containerRef.current
+
+    if (!containerCurrent) {
+      return
+    }
+
+    function handleTransitionEnd() {
+      setStatus(expanded ? 'open' : 'closed')
+    }
+
+    containerCurrent.addEventListener('transitionend', handleTransitionEnd)
+    setStatus(expanded ? 'opening' : 'closing')
+
+    return function navTransitionCleanup() {
+      if (containerCurrent) {
+        containerCurrent.removeEventListener('transitionend', handleTransitionEnd)
+      }
+    }
+  }, [expanded])
 
   return (
-    <StyledNav>
-      <StyledToggle
-        active={expanded}
-        type="button"
-        onClick={() => setExpanded(value => !value)}
-        aria-controls={componentId.current}
-      >
-        <div aria-hidden />
-        <span>Menu</span>
-      </StyledToggle>
-      <StyledNavCollapse height={expanded && navRef.current ? navRef.current.offsetHeight : 0}>
-        <StyledNavList id={componentId.current} aria-expanded={expanded} ref={navRef}>
-          {menu.map(item => (
-            <li key={item.href}>
-              <Link href={item.href} passHref>
-                <StyledNavLink active={isFirstBranchMatched(pagePath, item.href)}>
-                  {item.name}{' '}
-                  {item.external && <i className="fa fa-external-link" aria-label="will open in a new window" />}
-                </StyledNavLink>
-              </Link>
-            </li>
-          ))}
-        </StyledNavList>
-      </StyledNavCollapse>
+    <StyledNav status={status} ref={containerRef}>
+      <StyledNavList id={id} aria-expanded={status === 'open'} ref={navRef} tabIndex={-1}>
+        {menu.map(item => (
+          <li key={item.href}>
+            <Link href={item.href} passHref>
+              <StyledNavLink active={isFirstBranchMatched(pathname, item.href)}>
+                {item.name}
+                {/* TODO: Need new window icon */}
```

Comment:
Does this mean `master` won't be OK to go to prod if we merge?

Reply:
Ah, I see we aren't going into master yet

Reply:
Will chase up Kristy or try and find an SVG for this